### ISSUE TITLE
Be smarter about snapd's mntns re-use

### ISSUE THREAD

**Issue by stgraber** [lxc/lxd-pkg-snap#18]
snapd occasionally gives us a new mntns, when that happens, we find ourselves with the following problems:
 - Copies of filesystem mounts in the namespace of our previous processes (usually `lxcfs`)
 - Missing access to our own mounts (`shmounts`)

Short of being able to tell snapd not to do that, we need to add some logic to LXD to deal with the situation as best we can.

**Comment by stgraber** [comment]
This mostly happens when core changes, so we should be focusing on that.

**Comment by zertrin** [comment]
Glad to see this fixed, it's still biting us every few days...

Will it be backported to the 3.0 LTS branch?

**Comment by stgraber** [comment]
Yep, currently just in the edge channel for the latest track but will roll out to everyone over the next week or so.

**Comment by ser** [comment]
With 3.8 problem persists.
